crates/red_knot_ide/src/completion.rs

```diff
@@ -0,0 +1,43 @@
+use ruff_db::files::File;
+use ruff_db::parsed::parsed_module;
+use ruff_python_ast::visitor::source_order::{SourceOrderVisitor, TraversalSignal};
+use ruff_python_ast::AnyNodeRef;
+use ruff_text_size::TextSize;
+
+use crate::Db;
+
+pub struct Completion {
+    pub label: String,
+}
+
+pub fn completion(db: &dyn Db, file: File, _offset: TextSize) -> Vec<Completion> {
+    let parsed = parsed_module(db.upcast(), file);
+    identifiers(parsed.syntax().into())
+        .into_iter()
+        .map(|label| Completion { label })
+        .collect()
+}
+
+fn identifiers(node: AnyNodeRef) -> Vec<String> {
```

Comment:
This doesn't have to be as part of this PR because I think it would require fleshing out some API for it but we could do this by resolving the `semantic_index`, iterating over all scopes, and collecting the names from every scope. 

This would be a nice stepping stone to the next step where you'd identify in which scope you are and only suggest the names from that or its ancestor scopes.

Reply:
> we could do this by resolving the `semantic_index`, iterating over all scopes, and collecting the names from every scope.

we already have a query that basically does that in https://github.com/astral-sh/ruff/blob/main/crates/red_knot_python_semantic/src/semantic_index/re_exports.rs (but it only visits the global scope, currently at least)